pinot-compatibility-verifier/pom.xml

```diff
@@ -92,6 +92,11 @@
       <version>${project.version}</version>
       <type>test-jar</type>
     </dependency>
+    <dependency>
+      <groupId>org.testng</groupId>
+      <artifactId>testng</artifactId>
```

Comment:
This dependency was being used, but wasn't declared here and instead was get as a transitive dependency.

Reply:
Ideally this PR should be merged by the end of the week. Do you think it deserves to create another PR with this?

Reply:
I think it'd be quicker to land the changes in this PR if we find ways to cut them into smaller PRs gradually (while keeping this big one for cross reference). I'm more confident to stamp small PRs :)

Reply:
Ok! I added in https://github.com/apache/pinot/pull/10491